packages/router/router.ts

```diff
@@ -2977,22 +2985,14 @@ function processRouteLoaderData(
       if (result.headers) {
         loaderHeaders[id] = result.headers;
       }
-    } else if (isDeferredResult(result)) {
-      activeDeferreds && activeDeferreds.set(id, result.deferredData);
-      loaderData[id] = result.deferredData.data;
-      let init = result.deferredData.responseInit;
-      if (init) {
-        // Error status codes always override success status codes, but if all
-        // loaders are successful we take the deepest status code.
-        if (init.status != null && init.status !== 200 && !foundError) {
-          statusCode = init.status;
-        }
-        if (init.headers) {
-          loaderHeaders[id] = new Headers(init.headers);
-        }
-      }
```

Comment:
I aligned the `SuccessResult`/`DeferredResult` apis here for `statusCode` and `headers`